src/asl.ts

```diff
@@ -4739,40 +4789,138 @@ export namespace ASLGraph {
   };
 
   /**
-   * Visit each transition in each state.
-   * Use the callback to update the transition name.
+   * A {@link Choice} state that points to another {@link Choice} state can adopt the target state's
+   * choices and Next without adding an additional transition.
+   *
+   * 1
+   *    if a -> 2
+   *    if b -> 3
+   *    else -> 4
+   * 2
+   *    if c -> 3
+   *    else -> 4
+   * 3       - Task
+   * 4
+   *    if e -> 5
+   *    else -> 6
+   * 5       - Task
+   * 6       - Task
+   *
+   * =>
+   *
+   * 1
+   *    if a && c -> 3 (1 and 2)
+   *    if a && e -> 5 (1 and 4)
+   *    if b      -> 3
+   *    if e      -> 5 (4)
+   *    else      -> 6 (4's else)
+   * 2            - remove (if nothing else points to it)
+   * 3            - Task
+   * 4            - remove (if nothing else points to it)
+   * 5            - Task
+   * 6            - Task
    */
-  const visitTransition = (
-    state: State,
-    cb: (next: string) => string
-  ): State => {
-    if ("End" in state && state.End !== undefined) {
-      return state;
-    }
-    if (isChoiceState(state)) {
-      return {
-        ...state,
-        Choices: state.Choices.map((choice) => ({
-          ...choice,
-          Next: cb(choice.Next),
-        })),
-        Default: state.Default ? cb(state.Default) : undefined,
-      };
-    } else if ("Catch" in state) {
-      return {
-        ...state,
-        Catch: state.Catch?.map((_catch) => ({
-          ..._catch,
-          Next: _catch.Next ? cb(_catch.Next) : _catch.Next,
-        })),
-        Next: state.Next ? cb(state.Next) : state.Next,
+  const joinChainedChoices = (
+    startState: string,
+    stateEntries: [string, State][]
+  ) => {
```

Comment:
Please use `function declarations` instead of `const arrow function`.

Reply:
All of the functions in the namespaces where const arrow functions. I'll change them all to be functions.